app/models/manageiq/providers/kubevirt/infra_manager/event_catcher/runner.rb

```diff
@@ -0,0 +1,51 @@
+class ManageIQ::Providers::Kubevirt::InfraManager::EventCatcher::Runner < ManageIQ::Providers::Kubernetes::ContainerManager::EventCatcher::Runner
+  ENABLED_EVENTS = {
+    'VirtualMachine'         => %w(Created Started Migrated SuccessfulCreate SuccessfulDelete Killing),
+    'VirtualMachineInstance' => %w(Created Started Migrated SuccessfulCreate SuccessfulDelete Killing)
+  }
+
+  def event_monitor_handle
+    @event_monitor_handle ||= ManageIQ::Providers::Kubevirt::InfraManager::KubernetesEventMonitor.new(@ems)
+  end
+
+  def queue_event(event)
+    event_data = extract_event_data(event)
+    if !event_valid?(event_data)
+      _log.info "#{log_prefix} Skipping invalid event [#{event_data[:event_type]}]"
+      return
+    end
+
+    _log.info "#{log_prefix} Queuing event [#{event_data}]"
+    event_hash = ManageIQ::Providers::Kubevirt::InfraManager::EventParser.event_to_hash(event_data, @cfg[:ems_id])
+    EmsEvent.add_queue('add', @cfg[:ems_id], event_hash)
+  end
+
+  # Returns hash, or nil if event should be discarded.
+  def extract_event_data(event)
+    event_data = {
+      :timestamp => event.object.lastTimestamp,
+      :kind      => event.object.involvedObject.kind,
+      :name      => event.object.involvedObject.name,
+      :namespace => event.object.involvedObject.namespace,
+      :reason    => event.object.reason,
+      :message   => event.object.message,
+      :uid       => event.object.involvedObject.uid,
+      :event_uid => event.object.metadata.uid,
+    }
+
+    unless event.object.involvedObject.fieldPath.nil?
+      event_data[:fieldpath] = event.object.involvedObject.fieldPath
+    end
+
+    event_type_prefix = event_data[:kind].upcase
+
+    # Handle event data for specific entities
+    if event_data[:kind] == "VirtualMachine" || event_data[:kind] == "VirtualMachineKind"
+      event_data[:vm_name] = event_data[:name]
+    end
+
+    event_data[:event_type] = "#{event_type_prefix}_#{event_data[:reason].upcase}"
+
+    event_data
+  end
```

Comment:
Think all of this should live in the EventParser, the purpose of that class is to take the "native" event object and normalize it to our ems_event table